pkg/cmd/grafana-server/server.go

```diff
@@ -163,9 +163,21 @@ func (g *GrafanaServerImpl) Run() error {
 		})
 	}
 
-	sendSystemdNotification("READY=1")
+	defer func() {
+		g.log.Debug("Waiting on services...")
+		if waitErr := g.childRoutines.Wait(); waitErr != nil && !xerrors.Is(waitErr, context.Canceled) {
+			g.log.Error("A service failed", "err", waitErr)
+			if err == nil {
+				err = waitErr
+			}
+		}
+	}()
 
-	return g.childRoutines.Wait()
+	if err := sendSystemdNotification("READY=1"); err != nil {
+		g.log.Warn("Unable to send systemd notification", "err", err)
```

Comment:
I see this warning which is weird because Grafana doesn't run with systemd:
```
WARN[10-10|17:18:46] Unable to send systemd notification      logger=server err="NOTIFY_SOCKET environment variable empty or unset"
```

Not sure what the proper handling of this is? Maybe if `NOTIFY_SOCKET` environment variable is not set it shouldn't return an error, but not sure?